src/west/app/config.py

```diff
@@ -88,12 +93,18 @@ def do_add_parser(self, parser_adder):
             description=self.description,
             epilog=CONFIG_EPILOG)
 
-        parser.add_argument('-l', '--list', action='store_true',
-                            help='list all options and their values')
-        parser.add_argument('-d', '--delete', action='store_true',
-                            help='delete an option in one config file')
-        parser.add_argument('-D', '--delete-all', action='store_true',
-                            help="delete an option everywhere it's set")
+        group = parser.add_argument_group(
+            "action to perform (give at most one)"
+        ).add_mutually_exclusive_group()
+
+        group.add_argument('-l', '--list', action='store_true',
+                           help='list all options and their values')
+        group.add_argument('-d', '--delete', action='store_true',
+                           help='delete an option in one config file')
+        group.add_argument('-D', '--delete-all', action='store_true',
+                           help="delete an option everywhere it's set")
+        group.add_argument('-a', '--append', action='store_true',
+                           help='append to an existing value')
```

Comment:
Nice simplification thanks to `add_mutually_exclusive_group()` but I don't find it trivial enough to be combined with the addition of a feature that is not really trivial either. Please use (at least) 2 separate commits.